### ISSUE TITLE
False-positive: loop does not loop with multidimensional continue

### ISSUE THREAD

**Issue by rentalhost** [kalessil/phpinspectionsea#144]
I have the following code:

```php
$values = [
    [ 1, '', '' ],
    [ '', '', 6 ],
    [ 7, '', '' ],
    [ '', '', '' ],
    [ '', 8, '' ],
];

foreach ($values as $key => $subValues) {
    foreach ($subValues as $subValue) {
        if (is_int($subValue)) {
            printf("%s\n", $subValue);
            continue 2;
        }
    }

    printf("Found nothing at index %u\n", $key);
    break;
}
```

What is code does is: foreach() `$values` multidimensional array. Then foreach() this array elements too, and search for first ocurrency of an _int_, print, then `continue 2` (it's, make the first foreach() continue, instead of currently). If nothing is found, then it advances to next first foreach() statements. The result should be: "1, 6, 7. Found nothing at index 3" (then it should be break, and "8" will not be printed).

Then you can see that the first foreach() does looping, because it is looped by the `continue 2`. It is happening with for() too, or by using `return` instead of `break` on first looping.

The solution here is something like that: if your foreach() could be a "does not loop", check by existence of a internal `continue N`, where `N` is the distance between your looping and the internal loopings.

The code bellow should be valid too, for instance:

```php
foreach ($a as $av) {
    foreach ($b as $bv) {
        foreach ($c as $cv) {
            continue 3; // => Distance for first looping.
        }

        break; // Warning: really doesn't loop.
    }
    
    break; // Valid: does loop because continue 3.
}
```

PS.: add a label called "false-positive", please.

**Comment by kalessil** [comment]
false-positive = bug )

**Comment by kalessil** [comment]
@rentalhost : a fresh binary pushed, so you can give the fix a try.

**Comment by rentalhost** [comment]
@kalessil Works like a charm! Thanks!

**Comment by kalessil** [comment]
Super, thanks for checking ;)
